server/routes/posts/index.js

```diff
@@ -0,0 +1,61 @@
+const jwt = require('jsonwebtoken');
+const validateJWT = require('../../validation/validateJWT.js');
+const validatePost = require('../../validation/validatePost.js');
+const { initManagers } = require('../../db/initDB.js');
+const express = require('express');
+const app = module.exports = express();
+
+const { postManager } = initManagers();
+
+// submit a post
+app.post('/posts', async function (req, res, next) {
```

Comment:
Ideally, routes should include only routes with middleware declaration. Controllers should be located in the corresponding files for controllers.

Routes should look like (naming for schema and controller are up to you):
```
app.post('/api/v1/posts', checkJWT, celebrate(postsPostSchema), wrap(controller.postUpsert));
```
Explanation:
- the best practice for the naming of the route is to separate `api` requests and static ones, to include versioning to the routes
- you can pass several functions to create middlewares to check authorization and validate the data scheme with `celebrate` (library for validation)
Middleware is useful because in such a way you can easily understand which routes require authorization.
Without authorization:
```
app.get('/api/v1/posts', wrap(controller.getAllPosts));
```
- `wrap` here represents the error handler. It means that you won't need to use try-catch in every controller since they have an external try-catch already. And if you have `celebrate` validation, you would use separate error handler for 400 status separately for the wrapper for 500 status.
Something like:
```
module.exports = fn => (req, res, next) => {
    Promise.resolve(fn(req, res, next)).catch(next);
};
```